### PR TITLE
fix: clamp smart defaults to visible calendar range

### PR DESCRIPTION
**Pull request by joe-bor**
## Summary

- Add `getSmartDefaultTimes()` utility that clamps event default times to visible calendar range
- Events created outside 6 AM - 10 PM now default to 9 AM instead of current time
- Consolidate duplicated `START_HOUR` constants into shared `CALENDAR_START_HOUR`
- Handle edge case where rounding pushes time past the visible boundary

## Problem

Events created at times outside 6 AM - 11 PM are positioned off-screen on weekly/daily calendar views. This caused:

1. **Poor UX**: Users couldn't see newly created events without scrolling
2. **Flaky E2E tests**: CI running at early UTC hours (e.g., 4 AM) created events invisible on desktop views, causing test failures

### Root Cause

`getAddModeDefaults` in EventForm blindly used the current time, even if it fell outside the visible calendar grid (6 AM - 11 PM).

```typescript
// Before: Would create event at 4 AM if run at 4 AM
const now = new Date();
const roundedMinutes = Math.ceil(now.getMinutes() / 15) * 15;
// ... event created at current time, potentially off-screen
```

## Solution

Clamp smart default times to the visible calendar range (6 AM - 10 PM, leaving room for 1hr event before 11 PM).

```typescript
// After: Defaults to 9 AM when outside visible hours
const CALENDAR_START_HOUR = 6;  // 6 AM
const CALENDAR_END_HOUR = 22;   // 10 PM
const DEFAULT_HOUR = 9;         // Fallback

if (currentHour < CALENDAR_START_HOUR || currentHour >= CALENDAR_END_HOUR) {
  // Default to 9 AM - always visible
  workingDate.setHours(DEFAULT_HOUR, 0, 0, 0);
} else {
  // Within visible range - use rounded current time
  // ...

  // After rounding, if we've crossed the boundary, fall back
  if (workingDate.getHours() >= CALENDAR_END_HOUR) {
    workingDate.setHours(DEFAULT_EVENT_HOUR, 0, 0, 0);
  }
}
```

### Why This Works for All Views

- All calendar views use the same `EventForm` component via `EventFormModal`
- Smart defaults are centralized in one location
- Calendar grid boundaries are now shared constants used by views and defaults alike

### `src/lib/time-utils.ts`
- Added `getSmartDefaultTimes()` utility function with exported constants: `CALENDAR_START_HOUR` (6), `CALENDAR_END_HOUR` (22), `DEFAULT_EVENT_HOUR` (9)
- Includes post-rounding boundary check: times like 21:46 that round up to 22:00 fall back to 9 AM

### `src/lib/time-utils.test.ts`
- Added 15 unit tests covering all edge cases including the rounding-past-boundary scenario

### `src/components/calendar/components/event-form.tsx`
- Simplified `getAddModeDefaults` to use the new utility function (18 lines → 5 lines)

### `src/components/calendar/views/weekly-calendar.tsx` & `daily-calendar.tsx`
- Replaced duplicated local `START_HOUR = 6` with shared `CALENDAR_START_HOUR` import to prevent silent drift

### `CLAUDE.md`
- Added `getSmartDefaultTimes` and calendar boundary constants to the Date/Time Handling docs

## Test plan

- [x] All 401 unit tests pass (15 new for `getSmartDefaultTimes`)
- [x] All 4 E2E tests for "creates, views, edits, and deletes an event" pass (chromium, firefox, webkit, mobile-chrome)
- [x] Manual verification:
  - System time 4 AM → event defaults to 9 AM
  - System time 11 PM → event defaults to 9 AM
  - System time 9:59 PM → rounds to 10 PM, then falls back to 9 AM
  - System time 2 PM → event defaults to ~2 PM (current behavior preserved)

### CODE REVIEW

**Comment by joe-bor** [comment]
## PR Review: fix: clamp smart defaults to visible calendar range

### Summary
Adds a `getSmartDefaultTimes()` utility to `time-utils.ts` that clamps new event default times to the visible calendar grid (6 AM–10 PM), falling back to 9 AM when outside that range. Refactors `EventForm` to use this centralized utility.

### Verdict: Approve

Clean, well-structured change that follows established codebase patterns. The utility is well-tested, properly centralized, and solves a real UX + CI flakiness problem.

### Strengths
- **Follows architecture patterns**: Centralizes date/time logic in `time-utils.ts` as prescribed by CLAUDE.md
- **Excellent test coverage**: 13 unit tests covering normal cases, boundary conditions, and the invariant (end = start + 1hr)
- **Clean refactor**: EventForm goes from 18 lines of inline logic to a 5-line delegation — good separation of concerns
- **Testable design**: Accepting `now` as an optional parameter makes the function trivially testable without mocking `Date`
- **Atomic commits**: Three well-scoped conventional commits (feat → test → refactor)
- **Thorough PR description**: Clear problem statement, root cause, and solution explanation

### Issues (Must Fix)
None — this is ready to merge.

### Suggestions (Non-blocking)

1. **Consolidate duplicated `START_HOUR` constants**: Both `weekly-calendar.tsx:26` and `daily-calendar.tsx:29` define their own `const START_HOUR = 6`. Consider importing `CALENDAR_START_HOUR` from `time-utils.ts` in a follow-up PR to keep these in sync and avoid silent drift.

2. **Edge case: rounding past the visible boundary**: When the current time is 21:46–21:59, the function enters the "within visible hours" branch (since `currentHour` is 21) but rounds up to 22:00 or later, producing a start time at/past `CALENDAR_END_HOUR`. The event is still partially visible so this is minor, but worth noting. A future improvement could re-check after rounding:
   ```typescript
   // After rounding, if we've crossed the boundary, fall back
   if (workingDate.getHours() >= CALENDAR_END_HOUR) {
     workingDate.setHours(DEFAULT_EVENT_HOUR, 0, 0, 0);
   }
   ```

### Technical Debt
- The `START_HOUR = 6` duplication in `weekly-calendar.tsx` and `daily-calendar.tsx` should be consolidated with the new `CALENDAR_START_HOUR` export. Consider adding this to `TECHNICAL-DEBT.md`.

File: src/lib/time-utils.ts [hunk 1]
```diff
@@ -94,3 +94,51 @@ export function formatLocalDate(date: Date): string {
 export function parseLocalDate(dateStr: string): Date {
   return startOfDay(parseISO(dateStr));
 }
+
+/**
+ * Calendar grid visible hours constants.
+ * These match the weekly/daily view rendering boundaries.
```
Comment (joe-bor) [inline src/lib/time-utils.ts:100]: **Suggestion (non-blocking):** The calendar views (`weekly-calendar.tsx:26`, `daily-calendar.tsx:29`) each define their own `const START_HOUR = 6`. Consider importing `CALENDAR_START_HOUR` from here in a follow-up to keep these constants in sync.

File: src/lib/time-utils.ts [hunk 2]
```diff
@@ -94,3 +94,51 @@ export function formatLocalDate(date: Date): string {
...
+  const workingDate = new Date(now);
+  const currentHour = workingDate.getHours();
+
+  if (currentHour < CALENDAR_START_HOUR || currentHour >= CALENDAR_END_HOUR) {
+    // Outside visible hours - default to 9 AM
+    workingDate.setHours(DEFAULT_EVENT_HOUR, 0, 0, 0);
+  } else {
+    // Within visible hours - round up to next 15-min interval
+    const minutes = workingDate.getMinutes();
+    const roundedMinutes = Math.ceil(minutes / 15) * 15;
```
Comment (joe-bor) [inline src/lib/time-utils.ts:128]: **Minor edge case:** When time is 21:46–21:59, `currentHour` is 21 (passes the `< CALENDAR_END_HOUR` check), but rounding pushes the start to 22:00+. The event would start at/past the visible boundary. Consider adding a post-rounding check:
```typescript
// After rounding, if we've crossed the boundary, fall back
if (workingDate.getHours() >= CALENDAR_END_HOUR) {
  workingDate.setHours(DEFAULT_EVENT_HOUR, 0, 0, 0);
}
```
Not a blocker — the event is still partially visible and this is a narrow window.
